homeassistant/components/deconz/diagnostics.py

```diff
@@ -37,8 +38,42 @@ async def async_get_config_entry_diagnostics(
     }
     diag["alarm_systems"] = {k: v.raw for k, v in gateway.api.alarmsystems.items()}
     diag["groups"] = {k: v.raw for k, v in gateway.api.groups.items()}
-    diag["lights"] = {k: v.raw for k, v in gateway.api.lights.items()}
+
+    lights: dict[str, Any] = {k: v.raw for k, v in gateway.api.lights.covers.items()}
```

Comment:
This expanded section feels a bit worse than before. Is it possible to annotate that every value returned from `GroupedAPIItems.items()` is derived from `APIItem`? That way Mypy will know it has a `raw` attribute.

Reply:
This is a three step thing. The current refactoring focused on improving data management of the app and make it a bit more friendly to typing. Once all communication has been moved to the handlers I will simplify the data models as well and then this should hopefully go back to being the precious implementation. If you see some simple ways to attain this without what I just wrote I'd be very happy to take a chat on how that could be done. This is very much something I still have a lot to learn about.

Reply:
I reverted and added ignores for lights and sensors in the mean time

Reply:
I've tried playing around with typing in this code and `pydeconz`, and can't seem to fix the errors. I'm not sure if the annotations are wrong, or if there's a limitation being exposed in Mypy with some of the required circular imports in combination with Generics.

This is a tough one, and I'm also curious to hear of any solutions other than ignoring typing.

Reply:
I think the main problem is that a list can't be typed to hold multiple types. I need to change it to something like a dict or something. But this is the only place this is affected AFAIK so I wanted to work around it initially

Reply:
My hunch was wrong and I think I know the actual problem now. `gateway.api.lights.items()` returns a `dict[str, LightResources]`, but to iterate it you really want an `ItemsView[str, LightResources]`, which is more akin to an `Iterable[tuple[str, LightResources]]`. 

The old code was actually wrong (Mypy had flagged a genuine error) and so with the the current version of pydeconz (I'm testing with version 88), you'd need something like:

```py
    lights_dict = gateway.api.lights.items()
    diag["lights"] = {k: v.raw for k, v in lights_dict.items()}
```

But really, the `GroupedAPIItems.items()` method should return an `ItemsView` or an Iterable, like so:
```py
    def items(self) -> Iterable[tuple[str, DataResource]]:
        """Return items."""
        return itertools.chain.from_iterable(i.items() for i in self._items)
```

Note: I've tested that the above passes Mypy, but not that it actually works. You might wish to try it out if you use it.

Reply:
Thank you so much @chishm ! That fixed it, some other minor things broke which I fixed. I will put out a new release shortly, though this specific PR should be considered complete right now, going the second round once the lib gets the py.typed file included this will pass without an ugly ignore.